File: Packages/CaptureKit/Tests/CaptureKitTests/RecognitionResolverTests.swift
import TestKit
@testable import CaptureKit

let recognitionResolverTests: [TestCase] = [
    TestCase("qrBeatsText") { t in
        let r = RecognitionResolver.resolve(qrPayloads: ["https://example.com"],
                                            textLines: ["hello", "world"])
        t.equal(r, RecognitionResult.qr("https://example.com"))
    },
    TestCase("textLinesJoinWithNewlines") { t in
        let r = RecognitionResolver.resolve(qrPayloads: [], textLines: ["hello", "world"])
        t.equal(r, RecognitionResult.text("hello\nworld"))
    },
    TestCase("blankLinesAreDropped") { t in
        let r = RecognitionResolver.resolve(qrPayloads: [], textLines: ["", "hello", ""])
        t.equal(r, RecognitionResult.text("hello"))
    },
    TestCase("nothingIsNone") { t in
        t.equal(RecognitionResolver.resolve(qrPayloads: [], textLines: []),
                RecognitionResult.none)
        t.equal(RecognitionResolver.resolve(qrPayloads: [], textLines: ["", ""]),
                RecognitionResult.none)
    },
    TestCase("clipboardStrings") { t in
        t.equal(RecognitionResult.qr("x").clipboardString, "x")
        t.equal(RecognitionResult.text("y").clipboardString, "y")
        t.isNil(RecognitionResult.none.clipboardString)
    },
    TestCase("hudMessages") { t in
        t.equal(RecognitionResult.qr("x").hudMessage, "QR code copied")
        t.equal(RecognitionResult.text("abcd").hudMessage, "Text copied — 4 characters")
        t.equal(RecognitionResult.none.hudMessage, "No text found")
    },
]

File: windows/tests/BetterScreenshot.Tests/RecognitionResolverTests.cs
using BetterScreenshot.Capture;
using Xunit;

namespace BetterScreenshot.Tests;

public class RecognitionResolverTests
{
    [Fact]
    public void QrBeatsText()
    {
        var r = RecognitionResolver.Resolve(new[] { "https://example.com" }, new[] { "hello", "world" });
        Assert.Equal(RecognitionResult.Qr("https://example.com"), r);
    }

    [Fact]
    public void TextLinesJoinWithNewlines()
    {
        var r = RecognitionResolver.Resolve(Array.Empty<string>(), new[] { "hello", "world" });
        Assert.Equal(RecognitionResult.Text("hello\nworld"), r);
    }

    [Fact]
    public void BlankLinesAreDropped()
    {
        var r = RecognitionResolver.Resolve(Array.Empty<string>(), new[] { "", "hello", "" });
        Assert.Equal(RecognitionResult.Text("hello"), r);
    }

    [Fact]
    public void NothingIsNone()
    {
        Assert.Equal(RecognitionResult.None, RecognitionResolver.Resolve(Array.Empty<string>(), Array.Empty<string>()));
        Assert.Equal(RecognitionResult.None, RecognitionResolver.Resolve(Array.Empty<string>(), new[] { "", "" }));
    }

    [Fact]
    public void ClipboardStrings()
    {
        Assert.Equal("x", RecognitionResult.Qr("x").ClipboardString);
        Assert.Equal("y", RecognitionResult.Text("y").ClipboardString);
        Assert.Null(RecognitionResult.None.ClipboardString);
    }

    [Fact]
    public void HudMessages()
    {
        Assert.Equal("QR code copied", RecognitionResult.Qr("x").HudMessage);
        Assert.Equal("Text copied — 4 characters", RecognitionResult.Text("abcd").HudMessage);
        Assert.Equal("No text found", RecognitionResult.None.HudMessage);
    }
}
